eclair-core/src/main/scala/fr/acinq/eclair/db/pg/PgUtils.scala

```diff
@@ -181,10 +183,15 @@ object PgUtils extends JdbcUtils {
         }
       }
 
-      private def acquireExclusiveTableLock()(implicit connection: Connection): Unit = {
+      private def acquireExclusiveTableLock(lockTimeout: FiniteDuration)(implicit connection: Connection): Unit = {
         using(connection.createStatement()) {
           statement =>
-            statement.executeUpdate(s"LOCK TABLE $LeaseTable IN ACCESS EXCLUSIVE MODE NOWAIT")
+            // We use a timeout here, because we might not be able to get the lock right away due to concurrent access
+            // by other threads. That timeout gives time for other transactions to complete, then ours can take the lock
+            statement.executeUpdate(s"SET lock_timeout TO '${lockTimeout.toSeconds}s'")
```

Comment:
This is redone every time we re-acquire the lock, whereas we only need to do it once, right?
Does it make sense to do this somewhere else (when starting eclair) instead of redoing this (no-op) statement regularly?
If it's clearer to keep it here, don't bother fixing this.

Reply:
Done with 7a5bf8836.

It is less intrusive and more performant, but less obvious because the cause and the effect are separated in two different files. I guess it's okay.